### PR TITLE
fix(config-manager): Fixes polling interval and blocking timeout issues

### PR DESCRIPTION
**Pull request by rashidsp**
## Summary
- Validates upper limit of polling interval & blocking timeout.
- Adds relative log messages

## Test plan
- Adds/Updates unit tests

### CODE REVIEW

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 1]
```diff
@@ -185,18 +185,73 @@ def polling_interval(polling_interval)
...
 
-      # If polling interval is less than minimum allowed interval then set it to default update interval.
+      # If valid set given polling interval, default update interval otherwise.
 
-      if (polling_interval.is_a? Integer) && (polling_interval >= Helpers::Constants::CONFIG_MANAGER['MIN_UPDATE_INTERVAL'])
-        @polling_interval = polling_interval
+      if polling_interval.nil?
+        @logger.log(
+          Logger::DEBUG,
+          "Polling interval is not given. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
```
Comment (mikeng13) [inline lib/optimizely/config_manager/http_project_config_manager.rb]: Nit: ... not provided

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 2]
```diff
@@ -185,18 +185,73 @@ def polling_interval(polling_interval)
...
       end
 
-      @logger.log(
-        Logger::DEBUG,
-        "Invalid update_interval #{polling_interval} provided. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}"
-      )
-      @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
+      unless polling_interval.is_a? Integer
+        @logger.log(
+          Logger::DEBUG,
```
Comment (mikeng13) [inline lib/optimizely/config_manager/http_project_config_manager.rb]: This and the next lot should be error level logs

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 3]
```diff
@@ -185,18 +185,73 @@ def polling_interval(polling_interval)
...
+          "Blocking timeout is not given. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_BLOCKING_TIMEOUT']}."
+        )
+        @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_BLOCKING_TIMEOUT']
+        return
+      end
+
+      unless blocking_timeout.is_a? Integer
+        @logger.log(
+          Logger::DEBUG,
```
Comment (mikeng13) [inline lib/optimizely/config_manager/http_project_config_manager.rb]: Same thing regarding error level logs

File: lib/optimizely/config_manager/http_project_config_manager.rb [hunk 4]
```diff
@@ -185,18 +185,73 @@ def polling_interval(polling_interval)
...
+          "Polling interval '#{polling_interval}' has invalid type. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
+        )
+        @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
+        return
+      end
+
+      unless polling_interval.between?(Helpers::Constants::CONFIG_MANAGER['MIN_SECONDS_LIMIT'], Helpers::Constants::CONFIG_MANAGER['MAX_SECONDS_LIMIT'])
+        @logger.log(
+          Logger::DEBUG,
+          "Polling interval '#{polling_interval}' has invalid range. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}."
```
Comment (msohailhussain) [inline lib/optimizely/config_manager/http_project_config_manager.rb]: measurement unit would be good for all logs
`Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}. seconds`

Comment (mikeng13) [reply lib/optimizely/config_manager/http_project_config_manager.rb]: Yep we should specify the unit

**Pull request merged**